homeassistant/components/tado/binary_sensor.py

```diff
@@ -164,41 +162,22 @@ class TadoDeviceBinarySensor(TadoDeviceEntity, BinarySensorEntity):
 
     def __init__(
         self,
-        tado: TadoConnector,
+        coordinator: TadoDataUpdateCoordinator,
         device_info: dict[str, Any],
         entity_description: TadoBinarySensorEntityDescription,
     ) -> None:
         """Initialize of the Tado Sensor."""
         self.entity_description = entity_description
-        self._tado = tado
-        super().__init__(device_info)
+        self._tado = coordinator
```

Comment:
You're using a CoordinatorEntity, the coordinator is stored in `self.coordinator` by default